### ISSUE TITLE
`include` directive doesn't work as expected

### ISSUE THREAD

**Issue by flihp** [oxidecomputer/rfd-site#59]
We've been trying to get away from textual / tabular descriptions of PKI & certificate structures, preferring a KDL format understood by the `pki-playground` tool instead. I'm pulling the KDL into RFD 387 using the asciidoc `include` directive to make running the files through the `pki-playground` command line tool easier. I never thought to look till now but the RFD site doesn't render the KDL. It could be that I'm holding it wrong, but the site shows the `include:file/path.kdl` text instead of the contents of the file. I've been using `asciidoctor-pdf` to build locally and that works as expected. `include` looks like it might cause problems if we allow folks to pull in arbitrary URLs but files local to the repo would be sufficient for my needs.

**Comment by benjaminleonard** [comment]
So essentially what we're doing on the RFD site is returning the `.adoc` contents from the API and then running that through the AsciiDoc loader on the server. The issue then being that it's completely isolated from the context of the filesystem you have when running it locally. In this instance it has no reference to `platform-identity-csr.kdl`. The RFDs themselves exist in a database, so it's not working off a file system at all.

The image references we transform from a relative link to a signed URL from the GCP storage where they are kept. In theory I wonder if we might be able to chuck other non-image assets that sit in the RFD folder in there too. But it might not be worth the work to get it working, and whilst I understand it's a bit inconvenient to update manually it might still be better than the alternative.

Will defer to @augustuswm on what he thinks is best here.

**Comment by david-crespo** [comment]
A starting point: We could make it work locally, and then maybe the API would handle it by resolving references and inlining the content.

**Comment by benjaminleonard** [comment]
Inlining the content would be easier with this I think: https://github.com/oxidecomputer/react-asciidoc/pull/24

We could handle includes by post-processing the document object or we might not need to since it'd be running with the correct context. As it stands we'd have to do some sort of regex to find includes in the `adoc` file.

**Comment by augustuswm** [comment]
I think we can probably make this work. It would be simple enough to process the contents and handle files that match a known allowlist of file types.

**Comment by augustuswm** [comment]
Yeah, makes sense to me to move it to the `rfd-api` repo. I don't think we can or should support this on the frontend side when operating in remote mode. I think we could support it for local mode though. It would be similar parsing logic to what the API does.

I definitely want to support this on the API side, but I need to read up on how exactly it is supposed to operate.

**Comment by augustuswm** [comment]
I've added initial support for includes to the RFD API processing. We only support the simplest form which requires:

1. Includes exist entirely on their own line (as opposed to embedded in a sentence or other content)
2. No options may be passed to the include macro
  * `include::path/to/file.adoc()` will work but `include::path/to/file.adoc(indent=2)` will not
3. Paths must be relative to the location of the README.adoc
  * `include::path/to/file.adoc()` will work
  * `include::rfd/0001/path/to/file.adoc()` will not
  * `include::../path/to/file.adoc()` will not
  * `include::/rfd/0001/path/to/file.adoc()` will not
4. Paths must be written out, and not used attribute resolution to fill in variables. (This may or may not work depending on the RFD, so I'm marking as unsupported for now)
